nx/lib/nx/defn.ex

```diff
@@ -443,6 +443,34 @@ defmodule Nx.Defn do
     end
   end
 
+  @doc """
+  Wraps an anonymous function to return its underlying defn expression.
+
+  This function must be invoked only for debugging purposes only.
```

Comment:
```suggestion
  This function must be invoked for debugging purposes only.
```

We could also make it an admonition to bring more attention.